### PR TITLE
Authorization on server side, Alert message show up as modal 

### PR DESCRIPTION
**Pull request by Sherry2001**
Deployed: http://shershi-step-2020.appspot.com/
HTML Preview: https://htmlpreview.github.io/?https://github.com/Sherry2001/step/blob/master/portfolio/src/main/webapp/index.html

-Rica's comment on previous PR about moving email checks to server side. 

-Displaying delete action buffer as an alert message, with background darkened.

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/LogInServlet.java [hunk 1]
```diff
@@ -20,10 +20,14 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
...
     String email = null;
     String url;
+    Boolean access = false;
 
     if (userService.isUserLoggedIn()) {
       loggedIn = true;
       email = userService.getCurrentUser().getEmail();
+      if (email.equals("shershi@google.com") || email.equals("sherryshi2001@gmail.com") || email.equals("ricazhang@google.com") || email.equals("alfredh@google.com")) {
+        access = true;
+      }
```
Comment (ricazhang) [inline portfolio/src/main/java/com/google/sps/servlets/LogInServlet.java:30]: I would suggest storing a list of strings and check if the email is present in the list.

File: portfolio/src/main/java/com/google/sps/data/CheckLoginResponse.java [hunk 2]
```diff
@@ -6,10 +6,12 @@
   private final Boolean loggedIn;
   private final String url; 
   private final String email;
+  private final Boolean access;
 
-  public CheckLoginResponse(Boolean loggedIn, String url, String email) {
+  public CheckLoginResponse(Boolean loggedIn, String url, String email, Boolean access) {
```
Comment (ricazhang) [inline portfolio/src/main/java/com/google/sps/data/CheckLoginResponse.java:11]: Curious, why did you use the boolean object in all these cases?

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -43,10 +42,32 @@ function deleteBuffer() {
...
+/**
+ * Helper to show alert panel and blacked out background
+ */
+function showAlertMode(alertCard) {
+  alertCard.classList.add('active');
+  const backdrop = document.getElementById('backdrop');
+  backdrop.classList.add('active');
+  backdrop.addEventListener('click',() => {
+    hideAlert();
+  })
```
Comment (ricazhang) [inline portfolio/src/main/webapp/script.js:58]: I think this can be `addEventListener('click', hideAlert);`

**Review by Alfredhgoog — APPROVED** [review]

File: portfolio/src/main/java/com/google/sps/servlets/LogInServlet.java (hunk 1 above)
Comment (Sherry2001) [reply portfolio/src/main/java/com/google/sps/servlets/LogInServlet.java:30]: okay, changed!

File: portfolio/src/main/java/com/google/sps/data/CheckLoginResponse.java (hunk 2 above)
Comment (Sherry2001) [reply portfolio/src/main/java/com/google/sps/data/CheckLoginResponse.java:11]: as opposed to the primitive boolean? haha no reason- i just realized I forgot about the primitive boolean. I changed it.

File: portfolio/src/main/webapp/script.js (hunk 3 above)
Comment (Sherry2001) [reply portfolio/src/main/webapp/script.js:58]: Oh that works! I tried ('click', hideAlert()) before, but that just executes hideAlert immediately. I didn't know the () made a differencde.

**Review by ricazhang — APPROVED** [review]

**Pull request merged**